src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -93,12 +140,20 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
 
     auto allocator = zeroMemory::HostMemAllocator(backendPtr);
 
+    if (config.get<BATCH_MODE>() != ov::intel_npu::BatchMode::COMPILER) {
+        try {
+            _batch_size = get_batch_size(_metadata, executorInputDescriptors, executorOutputDescriptors);
```

Comment:
Can you consider return std::optional<size_t> from get_batch_size?

you can return nullopt in the case when get_batch_size cannot return a real size.

Reply:
I made this change but not sure we really need it. I would still keep the throw mechanism here for having a message if need more info when debugging such cases.

Reply:
You can still print messages (warning or error level) into logger when you think something bad happens and return std::nullopt.

I am still not sure if we need extra level of exceptions here...

Reply:
Btw, it is checked here if is 0 or not.